internal-enrichment/greynoise-vuln/src/main.py

```diff
@@ -0,0 +1,520 @@
+            )
+
+        if entity_type in scopes:
+            # OpenCTI entity information retrieval
+            stix_entity = data["stix_entity"]
+            opencti_entity = data["enrichment_entity"]
+            self.stix_objects = data["stix_objects"]
+
+            is_valid_max_tlp = self._extract_and_check_markings(opencti_entity)
+            if not is_valid_max_tlp:
+                raise ValueError(
+                    "[ERROR] Do not send any data, TLP of the observable is greater than MAX TLP, "
+                    "the connector does not has access to this observable, please check the group of the connector user"
+                )
+
+            # Extract Value from opencti entity data
+            opencti_entity_value = stix_entity["name"]
+
+            try:
+                # Get "CVE Context" GreyNoise API Response
+                # https://docs.greynoise.io/reference/get_v1-cve-cve-id
+                self.helper.connector_logger.info(
+                    f"Get CVE context for: {opencti_entity_value}"
+                )
+                session = GreyNoise(
+                    api_key=self.greynoise_key, integration_name=self.integration_name
+                )
+
+                json_data = session.cve(opencti_entity_value)
+
+                if "CVE not found" == json_data:
+                    stix_bundle = self.helper.stix2_create_bundle(self.stix_objects)
+                    bundles_sent = self.helper.send_stix2_bundle(stix_bundle)
+                    return (
+                        "[CONNECTOR] No CVE found. Original Vulnerability sent:  "
+                        + str(len(bundles_sent))
+                        + " stix bundle(s) for worker import"
+                    )
+                else:
+                    # Generate a stix bundle
+                    stix_bundle = self._generate_stix_bundle(json_data, stix_entity)
+
+                    # Send stix2 bundle
+                    bundles_sent = self.helper.send_stix2_bundle(stix_bundle)
+
+                return (
+                    "[CONNECTOR] Sent "
+                    + str(len(bundles_sent))
+                    + " stix bundle(s) for worker import"
+                )
+
+            except Exception as e:
+                # Handling other unexpected exceptions
+                raise ValueError(
+                    "[ERROR] Unexpected Error occurred :", {"Exception": str(e)}
+                )
+        else:
+            return self.helper.connector_logger.info(
+                "[INFO] The trigger does not concern the initial scope found in the config connector, "
+                "maybe choose a more specific filter in the playbook",
+                {"entity_id": data["entity_id"]},
+            )
+
+    # Start the main loop
+    def start(self):
+        self.helper.listen(message_callback=self._process_message)
+
+
+if __name__ == "__main__":
+    greyNoiseInstance = GreyNoiseVulnConnector()
+    greyNoiseInstance.start()
```

Comment:
Add a try except here in case the main process fail, we catch the issue:

```suggestion
    try:
        greyNoiseInstance = GreyNoiseVulnConnector()
        greyNoiseInstance.start()
    except Exception:
        traceback.print_exc()
        exit(1)
```